lib/txbr/metadata.rb

```diff
@@ -0,0 +1,30 @@
+module Txbr
+  class Metadata
+    ASSIGNMENTS = %w(project_slug resource_slug prefix)
+
+    attr_reader :project_slug, :resource_slug, :prefix
+
+    def initialize(options = {})
+      @project_slug = options.fetch('project_slug')
+      @resource_slug = options.fetch('resource_slug')
+      @prefix = options.fetch('prefix')
+    end
+
+    def ==(other)
+      project_slug == other.project_slug &&
+        resource_slug == other.resource_slug &&
+        prefix == prefix
+    end
+
+    def eql?(other)
+      hash == other.hash
+    end
+
+    def hash
```

Comment:
What's going on here?

Reply:
(Maybe a comment would be good)

Reply:
This is for calculating a [hash code](https://en.wikipedia.org/wiki/Hash_function) so Ruby can tell when two objects are "equal," i.e. they contain the same data. I just copied the technique Intellij uses when it auto-generates `hashcode` functions for Java classes. The hash function is supposed to be as unique as possible, and usually calculated on multiplying things by prime numbers. I'm not sure why, but apparently that works :)